### PR TITLE
Add F2Poly pretty Show instance

### PR DESCRIPTION
**Pull request by Multramate**

### CODE REVIEW

File: src/Data/Bit/F2Poly.hs [hunk 1]
```diff
@@ -132,6 +134,9 @@ instance Integral F2Poly where
   divMod = quotRem
   mod = rem
 
+instance Show F2Poly where
+  show = (:) '0' . (:) 'b' . flip (showIntAtBase 2 intToDigit) "" . toInteger
```
Comment (Bodigrim) [inline src/Data/Bit/F2Poly.hs]: In presence of `showIntAtBase` it would be more natural to define `showsPrec = const $ ...` instead of `show`.

Comment (Bodigrim) [reply src/Data/Bit/F2Poly.hs]: Actually, nevermind, it is fine as is.

Comment (Multramate) [reply src/Data/Bit/F2Poly.hs]: I will revert it then

**Comment by Bodigrim** [comment]
Thanks!

**Pull request merged**